lib/llm_composer.ex

```diff
@@ -78,15 +78,16 @@ defmodule LlmComposer do
   ## Returns
     - A tuple containing `:ok` with the response or `:error` if the model call fails.
   """
-  @spec run_completion(Settings.t(), messages(), LlmResponse.t() | nil) :: Helpers.action_result()
-  def run_completion(settings, messages, previous_response \\ nil) do
+  @spec run_completion(Settings.t(), messages(), LlmResponse.t() | nil, binary | nil) ::
+          Helpers.action_result()
+  def run_completion(settings, messages, previous_response \\ nil, api_key \\ nil) do
```

Comment:
Why don't we pass the optional api_key inside settings?

Reply:
+1

Reply:
Done! Not tested tho